### PR TITLE
Added option to disable some of the workers

### PR DESCRIPTION
**Pull request by amistad18**
This commit will add an option to disable some workers in config file (for example - if they are scanning forest area, with no pokemon spawns in there).

Also it will allow to display "circles" on the map for only chosen workers (also in config file), to make sure that "Show workers" might work also for those with over 100+ workers.

New config variables:
DISABLE_MARKERS = [0, 1, 2]
ENABLE_CIRCLES = [5,8,15]

Because of my city have some irregular shape, a lot of workers were scanning low populated or forest areas, that's because of the rectangle grid shape. After disabling almost half of the workers, I noticed a huge decrease at CPU usage :)
pokeminer_worker

And this is the web result:
pokeminer_web

Hope you'll find it useful

### CODE REVIEW

File: worker.py [hunk 1]
```diff
@@ -251,7 +256,7 @@ def get_status_message(workers, count, start_time, points_stats):
         '',
     ]
     previous = 0
-    for i in range(4, count + 4, 4):
+    for i in range(5, count + 5, 5):
```
Comment (VirtualSatai) [inline worker.py]: Maybe this should be a variable, in `config.py` which could be set, and have a reasonable default (which limits its width to 80 chars?).

Comment (amistad18) [inline worker.py]: Oh, I just forgot to remove it, that was just for a screenshot purpose, and I don't think that it would be useful in config for others.

**Comment by Primalpat-** [comment]
Will disabling workers allow their accounts to be reused in other workers?  If not, that would be a useful feature.. thought it seems complicated to implement.

Pokeminer would have to read the config, subtract out the disabled workers, and then spawn that many more enabled workers, which would potentially shift the layout of workers leading to disabled workers that should be enabled or enabled workers that should be disabled.

I guess I am just trying to figure out how to further optimize =)

**Comment by Sigi-cz** [comment]
https://github.com/modrzew/pokeminer/issues/61

**Comment by r0bodog** [comment]
How do I add this to the most recent release? I really need to disable the workers at sea.

**Comment by amistad18** [comment]
@r0bodog Not sure how to update this pull request, but here you have updated web.py and worker.py from v 0.5.0 with my changes, just copy them and replace with your current files
https://github.com/modrzew/pokeminer/pull/100/commits/137e39c6afa622f436f0ab2a2151b8b361388f92

@Primalpat- according to this part of the class Slave run method:
```
        service = config.ACCOUNTS[self.worker_no][2]
        while True:
            try:
                loginsuccess = self.api.login(
                    provider=service,
                    username=config.ACCOUNTS[self.worker_no][0],
                    password=config.ACCOUNTS[self.worker_no][1],
                )
```
It seams that it won't work that way so easy ... I was trying to change worker_no to account_no, and increment this value at the end of the loop, but it would need to refactor a bunch more of the code, and this is far beyond my python skills.

**Comment by YonderGod** [comment]
@amistad18 your latest version doesn't work. It disables circles, but not workers. I'm trying to figure it out myself, but just fyi.

**Comment by amistad18** [comment]
@YonderGod What do you mean? I'm running this on v0.5.0, with the code from patch that I pushed recently. Let me update to v0.5.1 and test it again. Also feel free to push your code if you already found some issue and fixed it :)

**Comment by YonderGod** [comment]
Maybe Im doing something wrong, but the 2nd commit was missing some code from your first one. I had to manually c&p and make a few changes to get it to work.

**Comment by paul30001** [comment]
How do I apply this as a Patch to my Pokeminer??

**Comment by modrzew** [comment]
@amistad18 could you rebase off most recent master? There's a lot of duplicated code here (you probably merged `forts` branch before it was in master).

**Comment by amistad18** [comment]
@modrzew sorry, but it looks I'm a git newbie, and I don't know how to update PR properly. I tried it with "Update for v0.5.0", but it looks like in the update I also put all your changes from v0.5.0 ...

Anyway, it looks like @YonderGod already did it:
https://github.com/modrzew/pokeminer/pull/238

It contains only the disable worker part of this PR, but after using open street maps, it looks like drawing circles is much more faster, so probably no need for the disabling circles part :)

btw. will this be still valid for v0.6 with coroutines? Or do you want to just merge that for those who wants to stay with python 2.7, before you go forward and drop support for anything less than v0.6?

**Comment by modrzew** [comment]
Moving to #238, then! Yes, it's still valid, and should be valid with v0.6.

**Pull request closed without merging**